wled00/FX_fcn.cpp

```diff
@@ -368,6 +369,7 @@ void Segment::beginDraw(uint16_t prog) {
     // minimum blend time is 100ms maximum is 65535ms
     #ifndef WLED_SAVE_RAM
     unsigned noOfBlends = ((255U * prog) / 0xFFFFU) - _t->_prevPaletteBlends;
+    if(noOfBlends > 255) noOfBlends = 255; // safety check
```

Comment:
No need for this check.

Reply:
If we trust in math then yes.
I.e. `prog` is a `_t->_progress` alias. `_progress` only increases with time. One exception is the one you found, when transition is restarted (`_start` and `_dur` change) to create a segment copy.